hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -159,18 +166,34 @@ public static HoodieTableMetaClient reload(HoodieTableMetaClient oldMetaClient)
    */
   private void readObject(java.io.ObjectInputStream in) throws IOException, ClassNotFoundException {
     in.defaultReadObject();
+    String basePathStr = in.readUTF();
+    String metaPathStr = in.readUTF();
+
     fs = null; // will be lazily initialized
+    basePath = new LazyCachingPath(basePathStr);
+    metaPath = new LazyCachingPath(metaPathStr);
   }
 
   private void writeObject(java.io.ObjectOutputStream out) throws IOException {
     out.defaultWriteObject();
+    out.writeBytes(basePath.toString());
+    out.writeBytes(metaPath.toString());
+  }
+
+  /**
+   * Returns base path of the table
+   */
+  public Path getBasePathV2() {
+    return basePath;
   }
 
   /**
    * @return Base path
+   * @deprecated please use {@link #getBasePathV2()}
    */
+  @Deprecated
```

Comment:
I see we are using getBasePath() in our baseRelation classes. do we need to fix them to getBasePathV2() ?

Reply:
We should slowly rollover all uses of `getBasePath` into `getBasePathV2` and then rename it